datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1193,9 +1193,9 @@ pub struct SubqueryAlias {
 #[derive(Clone)]
 pub struct Filter {
     /// The predicate expression, which must have Boolean type.
-    predicate: Expr,
+    pub predicate: Expr,
```

Comment:
I think this would allow side-stepping the schema validation in try_new?

Reply:
Yes, but I think it isn't a problem.
`Projection` `Aggregate` `Subquery` .... also is like this.

Reply:
Maybe we can at least document what invariants are required between fields

Reply:
> I think this would allow side-stepping the schema validation in try_new? I don't know how important this is

Yes, the motivation for the current design (introduced in https://github.com/apache/arrow-datafusion/pull/3796) was to prevent invalid filters from being created.